### PR TITLE
Add support for dependencies and security to paasta-tools

### PR DESCRIPTION
**Pull request by bchess**
Part of PAASTA-10896 @chriskuehl

### CODE REVIEW

File: paasta_tools/utils.py [hunk 1]
```diff
@@ -384,19 +384,51 @@ def check_disk(self):
...
+        security = self.config_dict.get('security')
+        if security is None:
+            return True, ''
+
+        outbound_firewall = security.get('outbound_firewall')
+        if outbound_firewall not in ('block', 'monitor'):
+            return False, 'Unrecognized outbound_firewall value "%s"' % outbound_firewall
+
+        unknown_keys = set(security.keys()) - {'outbound_firewall'}
+        if unknown_keys:
```
Comment (chriskuehl) [inline paasta_tools/utils.py:400]: nice

File: paasta_tools/utils.py [hunk 2]
```diff
@@ -384,19 +384,51 @@ def check_disk(self):
...
+            return False, 'Unrecognized keys in security dictionary: "%s"' % unknown_keys
```
Comment (chriskuehl) [inline paasta_tools/utils.py]: is there a better name than "keys" we could use? maybe "dependency group" or something?

There's lots of ways for yelpsoa-configs-related things to fail, and a lot of the times people are totally unable to parse the errors and just head into #paasta to ask. I think it's worth trying to be as clear as possible in the errors.

File: paasta_tools/utils.py [hunk 3]
```diff
@@ -442,6 +474,30 @@ def get_volumes(self, system_volumes):
         deduped = {v['containerPath'] + v['hostPath']: v for v in volumes}.values()
         return sort_dicts(deduped)
 
+    def get_dependencies_reference(self):
+        """Get the reference to an entry in dependencies.yaml
+
+        Defaults to None if not specified in the config.
+
+        :returns: A string specified in the config, None if not specified"""
+        return self.config_dict.get('dependencies_reference', None)
```
Comment (chriskuehl) [inline paasta_tools/utils.py]: None in .get(thing, None) is unnecessary

File: paasta_tools/utils.py [hunk 4]
```diff
@@ -442,6 +474,30 @@ def get_volumes(self, system_volumes):
...
+        :returns: A list of dictionaries specified in the dependencies_dict, None if not specified"""
+        return self.config_dict.get('dependencies', {}).get(self.get_dependencies_reference())
+
+    def get_outbound_firewall(self):
+        """Return 'block', 'monitor', or None as configured in security->outbound_firewall
+
+        Defaults to None if not specified in the config
+
+        :returns: A string specified in the config, None if not specified"""
+        return self.config_dict.get('security', {}).get('outbound_firewall', None)
```
Comment (chriskuehl) [inline paasta_tools/utils.py]: ditto

**Review by chriskuehl — APPROVED** [review]
lgtm

File: paasta_tools/utils.py (hunk 3 above)
Comment (bchess) [reply paasta_tools/utils.py]: ack

**Comment by bchess** [comment]
Ok this is mostly done - the only remaining TODO (prior to iptables integration) is identifying running chronos services

File: paasta_tools/utils.py (hunk 2 above)
Comment (bchess) [reply paasta_tools/utils.py]: ack

**Comment by bchess** [comment]
I've moved this to https://github.com/Yelp/paasta/pull/1234

**Pull request closed without merging**